sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasingSuite.scala

```diff
@@ -812,6 +812,33 @@ class NestedColumnAliasingSuite extends SchemaPruningTest {
     val expected3 = contact.select($"name").rebalance($"name").select($"name.first").analyze
     comparePlans(optimized3, expected3)
   }
+
+  test("SPARK-38530: Do not push down nested ExtractValues with other expressions") {
```

Comment:
Hmm, this test can pass in current master. Are you sure this can shows a bug?

Reply:
Sorry, when I looked at the code, it does not have the GetArrayStructFields(GetArrayStructFields()) check. I tweaked the test a bit so now it fails in the master.

Reply:
@viirya can you take a look again?